### PR TITLE
feat: support summary output for route/cluster/listener in hgctl gateway-config command (#995)

### LINKED ISSUE #995: Support summary output for route/cluster/listener in hgctl gateway-config command
## Why do you need it?
support summary output for route/cluster/listener in hgctl gateway-config command like istioctl as json/yaml format envoy config is not human readable.

## How could it be?
Just like istioctl proxy-config summary output format.

## Other related information
No

### PR DESCRIPTION
**Pull request by haifzhu**
### Ⅰ. Describe what this PR did
Support summary output for route/cluster/listener in hgctl gateway-config command 

### Ⅱ. Does this pull request fix one issue?

fixes #995

### CODE REVIEW

**Comment by johnlanni** [comment]
cc @2456868764

File: pkg/cmd/hgctl/config_cmd.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func newConfigCommand() *cobra.Command {
 	flags := cfgCommand.Flags()
 	options.AddKubeConfigFlags(flags)
 
-	cfgCommand.PersistentFlags().StringVarP(&output, "output", "o", "json", "One of 'yaml' or 'json'")
+	cfgCommand.PersistentFlags().StringVarP(&output, "output", "o", "short", "Output format: one of json|yaml|short")
```
Comment (johnlanni) [inline pkg/cmd/hgctl/config_cmd.go:55]: 默认值还是保持一致 用json吧

Comment (haifzhu) [reply pkg/cmd/hgctl/config_cmd.go:55]: 好

File: pkg/cmd/hgctl/config_cluster.go [hunk 2]
```diff
@@ -49,17 +50,23 @@ func runClusterConfig(c *cobra.Command, args []string) error {
 	if len(args) != 0 {
 		podName = args[0]
 	}
-	envoyConfig, err := config.GetEnvoyConfig(&config.GetEnvoyConfigOptions{
+	configWriter, err := config.GetEnvoyConfigWriter(&config.GetEnvoyConfigOptions{
 		PodName:         podName,
 		PodNamespace:    podNamespace,
 		BindAddress:     bindAddress,
 		Output:          output,
-		EnvoyConfigType: config.ClusterEnvoyConfigType,
```
Comment (haifzhu) [inline pkg/cmd/hgctl/config_cluster.go:57]: 这里类型不应该修改的

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**